text/dependencies/rfcs/0000-dependency-management-phase-one.md

```diff
@@ -0,0 +1,416 @@
+test:
+	@./test/test.sh $(tarballName) $(version)
+```
+
+#### 1. Version Retrieval (`make retrieve`)
+  * Description: Discovers dependency versions newer than `buildpack.toml`
+    versions (within constraints) and returns each with a set of generated
+    metadata in the form of a `metadata.json` file.
+  * Inputs:
+    * `buildpackTomlPath`  - the path to the  `buildpack.toml` file
+    * `output` - the path to a file into which an output metadata JSON will be
+      written
+  * Outputs:
+    * `output` - a `metadata.json` file containing entries for each new version
+      discovered at the `output` path.
+
+#### 2. Compilation (`make compile`)
+  * Description: Compiles a specific version of the dependency to the target
+    platform. Optional for dependencies that need it [Dependencies Directly
+    From
+    Upstream](https://github.com/paketo-buildpacks/rfcs/blob/cb117e4f45d2bc4e8d7551667100b5c8b9d7e89b/text/dependencies/rfcs/0000-dependency-management-phase-one.md#dependencies-directly-from-upstream)
+    section for more information. The future workflow that uses this code will
+    ensure that `outputDir` will exist and be writeable. For running locally,
+    this should be done by the user.
+  * Inputs:
+    * `version`  - the specific version to compile
+    * `outputDir` - the relative or absolute directory path for the output
+      tarball and other artifacts
+    * `target` - the target platform variant, which will only be used to
+      descriptively name the output artifact.
+  * Outputs:
+    The following files should be made available in the `outputDir`:
+    * Dependency artifact - There should be exactly one tarball, with a name of
+      `${dependency}-<description>.tgz`, where <description> can be any
+      combination of OS, architecture, target platform, version, etc that the
+      buildpack author desires. The dependency name will be highly dependent on
+      if the dependency needs to be compiled separately for different
+      OS/architecture combinations.
+    * Artifact SHASUM -  A file with the same name as the dependency artifact,
+      with a `sha256` suffix, containing the SHA256 of the compiled dependency.
+
+#### 3. Test Dependency (`make test`)
+  * Description: Performs some verification that the dependency is correctly
+    built and functions as expected. Optional for dependencies that have not
+    been compiled or processed.
+  * Inputs:
+    * `tarballPath` - the location of the tarball to test
+    * `version` - the dependency version
+
+### Transfer/Combine Version Retrieval and Metadata Generation
+
+The new  version retrieval code will merge together the job of discovering
+versions and getting metadata for each version. It will pick up newer versions
+than what the `buildpack.toml` contains, and outputs JSON-formatted metadata
+for each dependency. This code should be ported from the dep-server repository
+to the buildpack under a directory named `dependency/retrieval`, taking in the
```

Comment:
nit: "must be ported from dep server" contradicts what's said later, "feel free to reuse or rewrite the dep server..."

Reply:
fixed